prime-router/src/main/kotlin/fhirengine/utils/FHIRBundleHelpers.kt

```diff
@@ -51,4 +58,108 @@ object FHIRBundleHelpers {
 
         if (targetList.isNotEmpty()) provenanceResource.target.addAll(targetList)
     }
+
+    /**
+     * Gets all properties for a [Base] resource recursively and filters only its references
+     *
+     * @return a list of reference identifiers for a [Base] resource
+     *
+     */
+    fun Base.getResourceReferences(): List<String> {
```

Comment:
I made them extension functions because I thought it would be useful to know these functions are available when working with FHIR bundles without having to search through the code to see what's available.
I don't see any risks with doing this, we should just have to decide which one to use going forward if they add a similar function.